subsys/caf/modules/Kconfig.ble_adv

```diff
@@ -124,6 +124,50 @@ config CAF_BLE_ADV_GRACE_PERIOD
 	  and advertising is continued without Swift Pair data for additional
 	  time before the advertising is stopped.
 
+config CAF_BLE_ADV_ROTATE_RPA
+	bool "Rotate RPA together with advertising data"
+	depends on BT_PRIVACY
+	default y if BT_ADV_PROV_FAST_PAIR
+	help
+	  During active advertising, the module periodically triggers RPA
+	  rotation combined with advertising data update. This prevents
+	  automatic RPA rotations triggered by Zephyr in the background.
+
+	  Synchronizing RPA rotation with advertising data update is needed if
+	  an advertising provider uses a random value when generating data. In
+	  that case lack of synchronization could lead to compromising privacy.
+
+if CAF_BLE_ADV_ROTATE_RPA
+
+config CAF_BLE_ADV_ROTATE_RPA_TIMEOUT
+	int "RPA rotation timeout"
+	range 60 3600
+	default 720
+	help
+	  Period of time between subsequent RPA rotations triggered by module.
+
+config CAF_BLE_ADV_ROTATE_RPA_TIMEOUT_RAND
+	int "RPA rotation timeout randomization"
```

Comment:
Why is RPA timeout randomizaiton needed?

Reply:
To avoid tracking (if you know the exact time interval of RPA rotation, you can still track the device even it the address is rotated, because you know when to expect rotation).